homeassistant/components/lifx/coordinator.py

```diff
@@ -189,41 +192,54 @@ def async_get_entity_id(self, platform: Platform, key: str) -> str | None:
     async def _async_update_data(self) -> None:
         """Fetch all device data from the api."""
         async with self.lock:
-            if self.device.host_firmware_version is None:
-                self.device.get_hostfirmware()
-            if self.device.product is None:
-                self.device.get_version()
-            if self.device.group is None:
-                self.device.get_group()
-
-            response = await async_execute_lifx(self.device.get_color)
-
-            if self.device.product is None:
-                raise UpdateFailed(
-                    f"Failed to fetch get version from device: {self.device.ip_addr}"
-                )
-
-            # device.mac_addr is not the mac_address, its the serial number
-            if self.device.mac_addr == TARGET_ANY:
-                self.device.mac_addr = response.target_addr
-
-            if self._update_rssi is True:
-                await self.async_update_rssi()
-
-            # Update extended multizone devices
-            if lifx_features(self.device)["extended_multizone"]:
-                await self.async_get_extended_color_zones()
-                await self.async_get_multizone_effect()
-            # use legacy methods for older devices
-            elif lifx_features(self.device)["multizone"]:
-                await self.async_get_color_zones()
-                await self.async_get_multizone_effect()
-
-            if lifx_features(self.device)["hev"]:
-                await self.async_get_hev_cycle()
-
-            if lifx_features(self.device)["infrared"]:
-                await async_execute_lifx(self.device.get_infrared)
+            try:
+                if self.device.host_firmware_version is None:
+                    self.device.get_hostfirmware()
+                if self.device.product is None:
+                    self.device.get_version()
+                if self.device.group is None:
+                    self.device.get_group()
+
+                response = await async_execute_lifx(self.device.get_color)
+
+                if self.device.product is None:
+                    raise UpdateFailed(
+                        f"Failed to fetch get version from device: {self.device.ip_addr}"
+                    )
+
+                # device.mac_addr is not the mac_address, its the serial number
+                if self.device.mac_addr == TARGET_ANY:
+                    self.device.mac_addr = response.target_addr
```

Comment:
Aside: we should move this into the callback of `async_execute_lifx` so that it happens earlier.